src/elasticarray.jl

```diff
@@ -57,76 +67,139 @@ ElasticArray(A::AbstractArray{T,N}) where {T,N} = copyto!(ElasticArray{T,N}(unde
 Base.convert(::Type{T}, A::AbstractArray) where {T<:ElasticArray} = A isa T ? A : T(A)
 
 
+@inline function Base.similar(A::ElasticArray, ::Type{T}, dims::Dims{N}) where {T,N}
```

Comment:
To my knowledge, #6 addresses two separate issues. The first is supporting BLAS via `Base.unsafe_convert` while the second deals with whether `Base.similar(A::ElasticArrays, ...)` should return an `ElasticArray` or something else.

For proper CuArrays support, we do need to override `Base.similar(A::AbstractArray, ...)`, which is called in several places in e.g. `LinearAlgebra`. Without it, you'd get the following behavior:
```julia
julia> using ElasticArrays, CuArrays

julia> A = cu(ElasticArray{Float32}(undef, 2, 2));

julia> B = cu(ElasticArray{Float32}(undef, 2, 2));

julia> typeof(A * B)
Array{Float32,2}
```
That is, `A` and `B` are on the GPU but `A * B` is on the CPU, which is both slow and inconsistent with CuArrays:
```julia
julia> typeof(cu(rand(2,2)) * cu(rand(2,2)))
CuArray{Float32,2,Nothing}
```

That being said, we don't *have* to return an `ElasticArray`. The following would work:
```julia
Base.similar(A::ElasticArray, T::Type, dims::Dims) = similar(A.data, T, dims)
```

As for whether we *should* return an `ElasticArray` or not, I would argue that if I have `A::ElasticArray` and `B::ElasticArray` that it's reasonable to expect that `A * B` is also an `ElasticArray` (and is also consistent with e.g. StaticArrays/CuArrays etc.). But that also may require more effort on our part :).

Reply:
> As for whether we should return an ElasticArray [...] reasonable to expect that A * B is also an ElasticArray

That was my initial opinion, too, but @garrison argued otherwise (as to what to expect from `similar`) in #6.

But all that was a while ago - @garrison, do you want to weigh in here?

Reply:
Hm, if @garrison has no strong opinion here I suggest we merge this.